mavros/src/plugins/setpoint_actuator_control.cpp

```diff
@@ -0,0 +1,144 @@
+ * @author Marcel Stüttgen <stuettgen@fh-aachen.de>
+ *
+ * @addtogroup plugin
+ * @{
+ */
+/*
+ * Copyright 2015 Marcel Stüttgen.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ */
+
+#include <mavros/mavros_plugin.h>
+#include <mavros/setpoint_mixin.h>
+#include <pluginlib/class_list_macros.h>
+
+#include<std_msgs/Float32MultiArray.h>
+
+
+namespace mavplugin {
+
+/**
+ * @brief Setpoint actuator control plugin
+ *
+ * Send setpoint actuator control to FCU controller.
+ */
+class SetpointActuatorControlPlugin : public MavRosPlugin,
+                                      private TFListenerMixin<SetpointActuatorControlPlugin> {
+ public:
+  SetpointActuatorControlPlugin() :
+      uas(nullptr),
+      reverse_throttle(false)
+  { };
+
+  void initialize(UAS &uas_,
+                  ros::NodeHandle &nh,
+                  diagnostic_updater::Updater &diag_updater)
+  {
+    uas = &uas_;
+    sp_nh_ = ros::NodeHandle(nh, "setpoint_actuator_control");
+
+    sp_nh_.param("actuator_control/reverse_throttle", reverse_throttle, false);
+
+    controls_sub_ = sp_nh_.subscribe("controls", 10, &SetpointActuatorControlPlugin::control_cb, this);
+  }
+
+  const std::string get_name() const {
+    return "SetpointActuatorControl";
+  }
+
+  const message_map get_rx_handlers() {
+    return { /* Rx disabled */ };
+  }
+
+private:
+
+  friend class TFListenerMixin; //todo: need to find out what its name is mixin is needed for actuator controls
+  
+  UAS *uas;
+
+  ros::NodeHandle sp_nh_;
+  ros::Subscriber controls_sub_;
+
+  bool reverse_throttle;
+
+  /* -*- low-level send -*- */
+  /* message definiton here: https://pixhawk.ethz.ch/mavlink/#SET_ACTUATOR_CONTROL_TARGET */
+
+  void set_actuator_control_target(uint32_t time_boot_ms,
+                                   uint8_t group_mix,
+                                   uint8_t target_system,
+                                   uint8_t target_component,
+                                   float controls[8]) {
+    mavlink_message_t msg;
+
+    //todo: get correckt pack chan msg
+    /*
+    mavlink_msg_set_actuator_control_target_pack_chan(UAS_PACK_CHAN(uas), &msg, 
+                                                      time_boot_ms,
+                                                      UAS_PACK_TGT(uas),
```

Comment:
What about mavlink_msg_set_actuator_control_target_pack_chan ? i just guessed it and of course that does not work